### PR TITLE
Fix attachment annotation box render on non-PDF items

### PR DESCRIPTION
**Pull request by windingwind**
fix: #4477

### CODE REVIEW

**Comment by windingwind** [comment]
Just about to push that commit :) Thanks @AbeJellinek

File: chrome/content/zotero/elements/attachmentAnnotationsBox.js [hunk 1]
```diff
@@ -96,16 +96,27 @@
...
+				if (!imageAnnotationRendered
+					&& annotation.annotationType === 'image'
+					&& !await Zotero.Annotations.hasCacheImage(annotation)) {
+					try {
+						await Zotero.PDFWorker.renderAttachmentAnnotations(annotation.parentID);
+						imageAnnotationRendered = true;
+					}
+					catch (e) {
+						Zotero.debug(e);
+						throw e;
```
Comment (dstillman) [inline chrome/content/zotero/elements/attachmentAnnotationsBox.js:117]: Is this supposed to throw or continue?

File: chrome/content/zotero/elements/attachmentAnnotationsBox.js [hunk 2]
```diff
@@ -96,16 +96,27 @@
...
 			if (!this._section.open || this._annotationItems.length === 0) {
 				return;
 			}
 
 			this.hidden = false;
+			let imageAnnotationRendered = false;
 			for (let annotation of this._annotationItems) {
+				if (!imageAnnotationRendered
+					&& annotation.annotationType === 'image'
+					&& !await Zotero.Annotations.hasCacheImage(annotation)) {
```
Comment (dstillman) [inline chrome/content/zotero/elements/attachmentAnnotationsBox.js:110]: Indent these two lines

File: chrome/content/zotero/elements/attachmentAnnotationsBox.js [hunk 3]
```diff
@@ -96,16 +96,27 @@
...
+					try {
+						await Zotero.PDFWorker.renderAttachmentAnnotations(annotation.parentID);
+						imageAnnotationRendered = true;
+					}
+					catch (e) {
+						Zotero.debug(e);
```
Comment (dstillman) [inline chrome/content/zotero/elements/attachmentAnnotationsBox.js:116]: Log levels don't really matter, but `Zotero.debug(e, 1)` if we're really throwing and `Zotero.logError(e)` if we're continuing.

**Comment by windingwind** [comment]
Updated

**Pull request merged**